src/hub.js

```diff
@@ -1115,46 +1113,86 @@ document.addEventListener("DOMContentLoaded", async () => {
     return params;
   };
 
-  const migrateToNewReticulumServer = async deployNotification => {
-    // On Reticulum deploys, reconnect after a random delay until pool + version match deployed version/pool
-    console.log(`Reticulum deploy detected v${deployNotification.ret_version} on ${deployNotification.ret_pool}`);
-    clearInterval(retDeployReconnectInterval);
+  const reconnect = async ({ ret_pool, ret_version }) => {
+    invalidateReticulumMeta();
+    const { pool, version } = await getReticulumMeta();
+    return ret_pool === pool && ret_version === version;
+  };
 
-    setTimeout(() => {
-      const tryReconnect = async () => {
-        invalidateReticulumMeta();
-        const reticulumMeta = await getReticulumMeta();
-
-        if (
-          reticulumMeta.pool === deployNotification.ret_pool &&
-          reticulumMeta.version === deployNotification.ret_version
-        ) {
-          console.log("Reticulum reconnecting.");
-          clearInterval(retDeployReconnectInterval);
-          const oldSocket = retPhxChannel.socket;
-          const socket = await connectToReticulum(isDebug, oldSocket.params());
-          retPhxChannel = await migrateChannelToSocket(retPhxChannel, socket);
-          await hubChannel.migrateToSocket(socket, createHubChannelParams());
-          authChannel.setSocket(socket);
-          linkChannel.setSocket(socket);
-
-          // Disconnect old socket after a delay to ensure this user is always registered in presence.
-          setTimeout(() => {
-            console.log("Reconnection complete. Disconnecting old reticulum socket.");
-            oldSocket.teardown();
-          }, 10000);
+  const migrateToNewReticulumServer = async ({ ret_version, ret_pool }) => {
+    console.log(`[reconnect] Reticulum deploy detected v${ret_version} on ${ret_pool}.`);
+
+    const maxAttempts = 4;
+    let didReconnect = false;
+    let attempt = 0;
+    while (!didReconnect && attempt < maxAttempts) {
+      try {
+        // Add randomness to avoid flooding reticulum.
+        const delayMS = attempt * 3000 + Math.random() * 15000;
+        console.log(
+          `[reconnect] Attempting reconnect in ${Math.ceil(delayMS / 1000)} seconds.${
+            attempt ? ` (Attempt ${attempt + 1} of ${maxAttempts})` : ""
+          }`
+        );
+        await sleep(delayMS);
+        // Reconnect until reticulum meta matches expected version and pool.
```

Comment:
This comment made me think there would be a retry loop inside this function call. I honestly would probably just inline it, or move the loop into it, since its the core of what this loop is doing. Waiting for the version to be live.